### PR TITLE
Change Azure Directory -> Cloud. ALso changing Azure.Service type to …

### LINKED ISSUE #1075: Move Service interface outside of azure (cloud/ directory) package
/kind cleanup

**Describe the solution you'd like**
We have the [service](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/cloud/interfaces.go#L28) interface which describes the behavior of a reconciliation capz service. There are two kinds of services:

1. Individual service that reconciles one specific azure component (eg. virtualmachines.Service which only cares about vm reconciliation). These live inside `cloud/services` directory.
2. Composite Service (aka reconcilers) that reconciles more than one individual components (eg. azuremachineReconciler which reconciles all services required for an azuremachine). These are mostly inside `controllers/`

Right now, the `Service` interface is defined inside `cloud/` directory which makes it awkward for services that are outside of `cloud/services` to implement.

We'd like to move `Service` interface outside of `cloud/` to a common package accessible for all implementors.

**Anything else you would like to add:**

More discussion on this topic [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1053#discussion_r540551787)

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by surajrana93**
…Azure.Reconciler

 

 
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Clean up misnamed directory and Interace. Cleans up the code organization.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1075 

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [x] includes documentation

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by devigned** [comment]
/ok-to-test

Awesome! I'm super excited to see this refactor taking shape.

**Comment by shysank** [comment]
@surajrana93 Thanks for the PR! The tests are failing because the generated files are out of date, and a few linter errors. Try running `make generate` to generate the latest files, and `make lint` to see the the lint errors.

**Comment by CecileRobertMichon** [comment]
@surajrana93 thanks for the PR. When you get a chance, please rebase to fix the conflicts and sign the CNCF CLA so we can merge it.

**Comment by devigned** [comment]
Hi @surajrana93, just a gentle nudge. Are you interested in continuing this PR? If not, we are happy to help.

**Comment by surajrana93** [comment]
@devigned 
Hi David. Extremely sorry for the delay on this. I have rebased and got the Make Lint running on this change. I'll follow up on the other requirements as well.

Again I apologize for holding this up

**Comment by surajrana93** [comment]
I signed it

**Comment by devigned** [comment]
@surajrana93 no worries. We are grateful for your PR. Thank you for caring and wanting to make the community better through contribution!!

**Comment by surajrana93** [comment]
I signed it

**Comment by surajrana93** [comment]
@devigned  @shysank 
Any idea why the above 3 tests are failing

**Comment by devigned** [comment]
odd... I pulled down the PR and it built fine. I'm going to kick the build again. Perhaps, 410 GONE was intermittent.

**Review by devigned — CHANGES_REQUESTED** [review]
Please change https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/beb33be6f330c2e4ec09c7445f0d0c27834a58c6/.dockerignore#L6 to be `!/azure/**`

**Comment by CecileRobertMichon** [comment]
@surajrana93 please squash the two commits

**Comment by surajrana93** [comment]
@CecileRobertMichon I've squashed the comments.  I suppose the last test is failing because of the directory changes? Is there something to be done to fix that

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/assign @shysank @devigned

File: azure/services/privatedns/client.go
Comment (shysank) [inline azure/services/privatedns/client.go:25]: nit: do we need this prefix now that we have renamed it to azure?

File: azure/scope/cluster.go
Comment (devigned) [inline azure/scope/cluster.go:38]:
```suggestion
	"sigs.k8s.io/cluster-api-provider-azure/azure"
```

Please do this for the rest of the imports.

**Review by devigned — CHANGES_REQUESTED** [review]
Looks great outside of the superfluous import alias of `azure`.

**Comment by surajrana93** [comment]
@devigned @CecileRobertMichon removed the redundant package prefix

**Review by devigned — APPROVED** [review]
/lgtm

Great work!

**Comment by devigned** [comment]
One more rebase, please (i'm sorry).

**Comment by devigned** [comment]
/lgtm
/approve
Pending tests

**Comment by CecileRobertMichon** [comment]
/hold

@surajrana93 the commit history looks weird, there are two merge commits before the actual commit, can you please fix it so it looks clean?

@devigned @nader-ziada let's try to merge this before we merge anything else so we don't run into more conflicts (if it's ready to merge today and all tests are passing) since it's so hard to keep conflict free

**Comment by devigned** [comment]
Thank for catching that @CecileRobertMichon

**Comment by nader-ziada** [comment]
> /hold
> 
> @surajrana93 the commit history looks weird, there are two merge commits before the actual commit, can you please fix it so it looks clean?
> 
> @devigned @nader-ziada let's try to merge this before we merge anything else so we don't run into more conflicts (if it's ready to merge today and all tests are passing) since it's so hard to keep conflict free

yes, lets get this merged soon, I'm looking forward to rebasing the v1alpha4 PR on this change 😃

**Comment by CecileRobertMichon** [comment]
I went ahead and rebased to get this in since @surajrana93 is in a different time zone

@devigned @nader-ziada I'll let you lgtm to make sure I didn't make any rebase mistakes (it was a clean rebase, no conflicts)

/hold cancel

**Pull request merged**
